couchbase/datadog_checks/couchbase/couchbase.py

```diff
@@ -402,3 +408,66 @@ def extract_seconds_value(self, value):
             unit = 'us'
 
         return float(val) / TO_SECONDS[unit]
+
+    def _collect_index_stats_metrics(self):
+        url = urljoin(self._index_stats_url, INDEX_STATS_METRICS_PATH)
+        try:
+            data = self._get_stats(url)
+        except requests.exceptions.RequestException as e:
+            msg = "Error accessing the Index Statistics endpoint: %s: %s" % (url, str(e))
+            self.log.debug(str(e))
+            self.service_check(INDEX_STATS_SERVICE_CHECK_NAME, AgentCheck.CRITICAL, self._tags, msg)
+            return
+
+        self.service_check(INDEX_STATS_SERVICE_CHECK_NAME, AgentCheck.OK, self._tags)
+
+        for keyspace in data:
+            if keyspace == "indexer":
+                # The indexer object provides metric about the index node
+                for mname, mval in data.get(keyspace).items():
+                    self._submit_index_node_metrics(mname, mval, self._tags)
+            else:
+                index_tags = self._extract_index_tags(keyspace) + self._tags
+                for mname, mval in data.get(keyspace).items():
+                    self._submit_per_index_metrics(mname, mval, index_tags)
+
+    def _extract_index_tags(self, keyspace):
+        # Index Keyspaces can come in different formats:
+        # partition, bucket:index_name, bucket:collection:index_name, bucket:scope:collection:index_name
+        # For variations missing the scope and collection, they refer to the default scope and collection respectively
+        # https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/createprimaryindex.html#keyspace-ref
+        tag_arr = keyspace.split(":")
+        if len(tag_arr) == 1:
+            partition = tag_arr[0]
```

Comment:
are there any docs that you can link that describe this case, when partition is the keyspace?

Reply:
Left a comment in which a use case exist in which the `partition-num` would be a keyspace. But we won't come across that in this iteration of the check. I converted it into a catch all incase the keyspace has more separators then we expect as well.